### PR TITLE
Add more utilities including install and symbolic link

### PR DESCRIPTION
**Pull request by izgzhen**

### CODE REVIEW

File: cfg/system.config.in [hunk 1]
```diff
@@ -117,3 +118,17 @@ ffi-lib-dir       = @FFILibDir@
 #=======================
 
 with-libdw = @UseLibdw@
+
+# Installation:
+#=======================
+
+prefix          = @prefix@
```
Comment (snowleopard) [inline cfg/system.config.in]: Is `prefix` used only for installation? If yes, I guess `install-prefix` would be easier to understand.

File: cfg/system.config.in [hunk 2]
```diff
@@ -117,3 +118,17 @@ ffi-lib-dir       = @FFILibDir@
...
+bindir          = @prefix@/bin
```
Comment (snowleopard) [inline cfg/system.config.in]: ... and `install-bin-dir`, etc.?

File: src/Oracles/Config/Setting.hs [hunk 3]
```diff
@@ -51,6 +52,16 @@ data Setting = BuildArch
              | GmpLibDir
              | IconvIncludeDir
              | IconvLibDir
+             | Prefix
```
Comment (snowleopard) [inline src/Oracles/Config/Setting.hs]: Same here: let's keep `Install` prefix for all install-related settings to keep the code a bit easier to understand.

File: src/Oracles/Config/Setting.hs [hunk 4]
```diff
@@ -173,3 +194,26 @@ cmdLineLengthLimit = do
         (False, True) -> 200000
         -- On all other systems, we try this:
         _             -> 4194304 -- Cabal library needs a bit more than 2MB!
+
+-- | https://github.com/ghc/ghc/blob/master/mk/config.mk.in#L241
+relocatableBuild :: Action Bool
+relocatableBuild = windowsHost
+
+prefix, libdir, bindir, datarootdir :: Action String
```
Comment (snowleopard) [inline src/Oracles/Config/Setting.hs]: ... and here: `installPrefix`, etc.

Comment (snowleopard) [inline src/Oracles/Config/Setting.hs]: By the way, do we really need separate `Action`s for this? I think we could just use `setting InstallPrefix` when needed. I doubt it's used often enough to justify an extra abstraction layer.

File: src/Oracles/Config/Setting.hs [hunk 5]
```diff
@@ -173,3 +194,26 @@ cmdLineLengthLimit = do
...
+datarootdir = setting DataRootDir
+docdir = do
+  version <- setting ProjectVersion
+  (-/- ("doc/ghc-" ++ version)) <$> datarootdir
+
+-- mk/install.mk: 108
+ghclibdir :: Action String
+ghclibdir = do
+  r <- relocatableBuild
+  libdir <- setting LibDir
```
Comment (snowleopard) [inline src/Oracles/Config/Setting.hs]: In fact, you are using `setting LibDir` instead of `libdir` youself! :-)

Comment (izgzhen) [inline src/Oracles/Config/Setting.hs]: Ah, you caught it!

File: cfg/system.config.in (hunk 1 above)
Comment (izgzhen) [inline cfg/system.config.in]: Well this is a good point, although `prefix` is already a commonly used term, but it is a bit confusing in itself. So `install-prefix` is probably better

**Comment by izgzhen** [comment]
Fixed

**Comment by izgzhen** [comment]
Ready for merging as well.

**Comment by snowleopard** [comment]
Thanks, merged!

Can we have a summary somewhere, explaining the difference between all `InstallPrefix`-like settings? I think I have a vague understanding about them, but a newcomer to the code base would benefit from some explanation. I guess the best place for it would be `Rules.Install`.

**Comment by izgzhen** [comment]
> Can we have a summary somewhere, explaining the difference between all InstallPrefix-like settings? 

Sure

**Pull request merged**